src/pudl/io_managers.py

```diff
@@ -31,56 +34,82 @@
+    # @io_manager constructor of this, see "def pudl_io_manager".
+    write_to_parquet: bool
+    """If true, data will be written to parquet files."""
 
+    read_from_parquet: bool
+    """If true, data will be read from parquet files instead of sqlite."""
 
-class ForeignKeyErrors(SQLAlchemyError):  # noqa: N818
-    """Raised when data in a database violate multiple foreign key constraints."""
+    def __init__(self, write_to_parquet: bool = False, read_from_parquet: bool = False):
+        """Creates new instance of mixed format pudl IO manager.
 
-    def __init__(self, fk_errors: list[ForeignKeyError]):
-        """Initialize a new ForeignKeyErrors object."""
-        self.fk_errors = fk_errors
+        By default, data is written and read from sqlite, but experimental
+        support for writing and/or reading from parquet files can be enabled
+        by setting the corresponding flags to True.
 
-    def __str__(self):
-        """Create string representation of ForeignKeyErrors object."""
-        fk_errors = [str(x) for x in self.fk_errors]
-        return "\n".join(fk_errors)
+        Args:
+            write_to_parquet: if True, all data will be written to parquet
+                files in addition to sqlite.
+            read_from_parquet: if True, all data reads will be using
+                parquet files as source of truth. Otherwise, data will be
+                read from the sqlite database.
+        """
+        if read_from_parquet and not write_to_parquet:
+            raise RuntimeError(
+                "read_from_parquet cannot be set when write_to_parquet is False."
+            )
+        self.write_to_parquet = write_to_parquet
+        self.read_from_parquet = read_from_parquet
+        self._sqlite_io_manager = PudlSQLiteIOManager(
+            base_dir=PudlPaths().output_dir,
+            db_name="pudl",
+        )
+        self._parquet_io_manager = PudlParquetIOManager()
+        if self.write_to_parquet or self.read_from_parquet:
+            logger.warning(
+                f"pudl_io_manager: experimental support for parquet enabled. "
+                f"(read={self.read_from_parquet}, write={self.write_to_parquet})"
+            )
 
-    def __iter__(self):
-        """Iterate over the fk errors."""
-        return self.fk_errors
+    def handle_output(
+        self, context: OutputContext, obj: pd.DataFrame | str
+    ) -> pd.DataFrame:
+        """Passes the output to the appropriate IO manager instance."""
+        self._sqlite_io_manager.handle_output(context, obj)
+        if self.write_to_parquet:
+            self._parquet_io_manager.handle_output(context, obj)
 
-    def __getitem__(self, idx):
-        """Index the fk errors."""
-        return self.fk_errors[idx]
+    def load_input(self, context: InputContext) -> pd.DataFrame:
+        """Reads input from the appropriate IO manager instance."""
+        if self.read_from_parquet:
+            return self._parquet_io_manager.load_input(context)
```

Comment:
It'd be helpful to add a comment or docstring that says we prefer to load inputs from parquet files because of speed and type preservation.

Reply:
Good call, I added to the argument description in the docstring

Reply:
Hopefully this ends up being true. 🤞🏼